File: backend/Model/Shipment.cs
using System.Text.Json.Serialization;

namespace CosmoCargo.Model;

public class Shipment
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid? PilotId { get; set; }
    public ShipmentContact Sender { get; set; } = new();
    public ShipmentContact Receiver { get; set; } = new();

    // Package information
    public decimal Weight { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Priority { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool HasInsurance { get; set; }

    // Status and tracking
    public ShipmentStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    [JsonIgnore] public virtual User Customer { get; set; } = null!;

    [JsonIgnore] public virtual User? Pilot { get; set; }
}

File: frontend/src/model/shipment.ts
import { ShipmentStatus } from './types';
import { BaseEntity } from './common';
import User from './user';

export interface ShipmentContact {
    name: string;
    email: string;
    planet: string;
    station: string;
}

export default interface Shipment extends BaseEntity {
    origin: ShipmentContact;
    destination: ShipmentContact;
    pilotName: string;
    customerId: string;
    pilotId?: string | null;
    sender: ShipmentContact;
    receiver: ShipmentContact;
    weight: number;
    category: string;
    priority: string;
    description?: string | null;
    hasInsurance: boolean;
    status: ShipmentStatus;
    createdAt: string;
    updatedAt: string;
    pilot?: User | null;
}
